### PR TITLE
Make it work both with Python2 and Python3

### PR DESCRIPTION
**Pull request by mike-fabian**
With this change (and some changes I did to the python code in https://github.com/googlei18n/noto-emoji.git I was able to build the "NotoColorEmoji.ttf" font using Python3.

### CODE REVIEW

File: nototools/generate_sample_text_html.py [hunk 1]
```diff
@@ -62,18 +62,18 @@ def generate_table(filename):
...
         cols.append(bcp_no_script)
         cols.append(bcp_en)
         cols.append(sample_type)
         cols.append(sample_text)
-        print >> f, '<td>'.join(cols)
-      print >> f, '<tr><td colspan=4>&nbsp;'
-    print >> f, '</table>'
-    print >> f, _HTML_FOOTER
+        f.write('<td>'.join(cols) + '\n')
+      f.write('<tr><td colspan=4>&nbsp;')
```
Comment (dougfelt) [inline nototools/generate_sample_text_html.py]: why no '\n' added in these lines?

File: nototools/noto_cmap_reqs.py [hunk 2]
```diff
@@ -3285,10 +3285,10 @@ def _dump_primaries():
   for block in unicode_data.block_names():
     block_range = unicode_data.block_range(block)
     primary_script = _primary_script_for_block(block)
-    print '%13s %6s %s' % (
+    print('%13s %6s %s\n' % (
```
Comment (dougfelt) [inline nototools/noto_cmap_reqs.py]: It seems the '\n' is not wanted here?  Print already issues a newline, right?

File: nototools/sync_repos.py [hunk 3]
```diff
@@ -84,17 +84,17 @@ def noto_checkout(
       failed_tags.append('%s: %s' % (r, t))
 
   if failed_tags:
-    print >> sys.stderr, 'failed to find:\n  %s' % '\n  '.join(failed_tags)
+    sys.stderr.write('failed to find:\n  %s' % '\n  \n'.join(failed_tags))
```
Comment (dougfelt) [inline nototools/sync_repos.py]: The added newline is in the wrong place.

File: nototools/unicode_data.py [hunk 4]
```diff
@@ -784,7 +784,7 @@ def _read_emoji_data(lines):
       continue
 
     # discourage lots of redundant copies of seq_type
```
Comment (dougfelt) [inline nototools/unicode_data.py]: delete comment, no longer applies?

File: nototools/generate_sample_text_html.py (hunk 1 above)
Comment (dougfelt) [reply nototools/generate_sample_text_html.py]: meant to make that a review comment

Comment (mike-fabian) [reply nototools/generate_sample_text_html.py]: Oh, you are right! adding these '\n' ...

File: nototools/noto_cmap_reqs.py (hunk 2 above)
Comment (mike-fabian) [reply nototools/noto_cmap_reqs.py]: You  are right! Sorry!

File: nototools/sync_repos.py (hunk 3 above)
Comment (mike-fabian) [reply nototools/sync_repos.py]: Yes!

File: nototools/unicode_data.py (hunk 4 above)
Comment (mike-fabian) [reply nototools/unicode_data.py]: Yes, no longer applies when  the intern() is gone!

**Comment by PureTryOut** [comment]
So this will close #472 when merged then.

**Comment by mike-fabian** [comment]
Do I need to do something about the “Need a CLA for one or more commit authors”?

The commits have mfabian@redhat.com as the author e-mail, should I change that to my Google mail maiku.fabian@gmail.com because I used that to sign the CLA? I tried to add mfabian@redhat.com as an additional address to my google account, but that didn’t work because it is used in my work account and apparently one cannot add the same address to another google account.

**Comment by dougfelt** [comment]
Hmmm, I'm not sure about workarounds for this.  If you can change the
commit address that would likely work.

**Comment by mike-fabian** [comment]
> Hmmm, I'm not sure about workarounds for this. If you can change the commit address that would likely work.

I have changed the commit address.

**Comment by RoelN** [comment]
I can't get `pip3 install -r requirements.txt` to install properly. I'm on macOS High Sierra. It errors out on:

  ```
Building wheels for collected packages: pyclipper
  Building wheel for pyclipper (setup.py) ... error
  [...]
    2 warnings and 1 error generated.
    error: command 'clang' failed with exit status 1
    ----------------------------------------
    ERROR: Failed building wheel for pyclipper
    Running setup.py clean for pyclipper
    Failed to build pyclipper
  [...]
  ```

and then:

```
ERROR: ufo2ft 2.9.1 has requirement booleanOperations>=0.8.2, but you'll have booleanoperations 0.7.0 which is incompatible.
ERROR: ufo2ft 2.9.1 has requirement defcon>=0.6.0, but you'll have defcon 0.3.1 which is incompatible.
ERROR: mutatormath 2.1.2 has requirement defcon>=0.3.5, but you'll have defcon 0.3.1 which is incompatible.
ERROR: fontmake 1.10.1 has requirement booleanOperations>=0.8.2, but you'll have booleanoperations 0.7.0 which is incompatible.
ERROR: fontmake 1.10.1 has requirement defcon[lxml]>=0.6.0, but you'll have defcon 0.3.1 which is incompatible.
```

I eventually managed to get al requirements in order by manual installs through pip3 and brew. I was able to run:

`sudo python3 setup.py develop`

So while it's working now (a Noto Emoji build is happily purring in the background), I'm not sure if I'm doing something wrong, the requirements.txt need to be modified or something else is wrong. Any advice?

**Comment by RoelN** [comment]
If nobody else is running into problems, bite the bullet and go full python 3? @dougfelt

**Comment by RoelN** [comment]
https://github.com/googlefonts/noto-emoji/pull/267 is waiting for this PR. Can we merge this?

File: nototools/unicode_data.py [hunk 5]
```diff
@@ -783,8 +783,7 @@ def _read_emoji_data(lines):
     if m.group(2):
       continue
 
-    # discourage lots of redundant copies of seq_type
-    seq_type = intern(m.group(3).strip().encode('ascii'))
```
Comment (anthrotype) [inline nototools/unicode_data.py]: FWIW the old `intern` builtin was moved to `sys.intern`: https://docs.python.org/3/library/sys.html?highlight=sys.intern#sys.intern

File: nototools/collect_cldr_punct.py [hunk 6]
```diff
@@ -99,7 +99,7 @@ def _collect_script_to_punct(files):
       script = cldr_data.get_likely_script(filename)
       if script == 'Zzzz':
         if filename != 'root':
-          print >> sys.stderr, 'no script for %s' % filename
+          sys.stderr.write('no script for %s\n' % filename)
```
[the remark this answers is not in the document]
Comment (plugwash) [reply nototools/collect_cldr_punct.py]: sys.stderr.write(foo) works in both python2 and python3, print(foo, file=sys.stderr) only works in python3.

[the remark this answers is not in the document]
Comment (anthrotype) [reply nototools/collect_cldr_punct.py]: You need to `from __future__ import print_function`.
If we haven't done that yet, we should make sure to add this as well as the other two canonical future imports (`absolute_import`, `division`) to all modules.

**Pull request merged**